packages/eas-cli/src/commands/build/version/sync.ts

```diff
@@ -0,0 +1,174 @@
+import { findProjectRootAsync, getProjectIdAsync } from '../../../project/projectUtils';
+import {
+  ensureRemoteVersionPolicyAsync,
+  validateAppConfigForRemoteVersionPolicyAsync,
+} from '../../../project/remoteVersionSource';
+import { resolveWorkflowAsync } from '../../../project/workflow';
+import { getProfilesAsync } from '../../../utils/profiles';
+
+interface SyncContext<T extends Platform> {
+  projectDir: string;
+  exp: ExpoConfig;
+  workflow: Workflow;
+  profile: BuildProfile<T>;
+  buildVersion: string;
+}
+
+export default class BuildVersionSyncView extends EasCommand {
+  public static description = 'Update a version in native code with a value stored on EAS servers';
+  static hidden = true;
+
+  public static flags = {
+    platform: Flags.enum({
+      char: 'p',
+      options: ['android', 'ios', 'all'],
+    }),
+    profile: Flags.string({
+      description:
+        'Name of the build profile from eas.json. Defaults to "production" if defined in eas.json.',
+      helpValue: 'PROFILE_NAME',
+    }),
+  };
+
+  public async runAsync(): Promise<void> {
+    const { flags } = await this.parse(BuildVersionSyncView);
+    const projectDir = await findProjectRootAsync();
+
+    const requestedPlatform = await selectRequestedPlatformAsync(flags.platform);
+    const easJsonReader = new EasJsonReader(projectDir);
+    await ensureRemoteVersionPolicyAsync(projectDir, easJsonReader);
+
+    const platforms = toPlatforms(requestedPlatform);
+    const buildProfiles = await getProfilesAsync({
+      type: 'build',
+      easJsonReader,
+      platforms,
+      profileName: flags.profile ?? undefined,
+    });
+    for (const profileInfo of buildProfiles) {
+      const exp = getExpoConfig(projectDir, { env: profileInfo.profile.env });
+      const projectId = await getProjectIdAsync(exp);
+      await validateAppConfigForRemoteVersionPolicyAsync(exp);
+      const platformDisplayName = appPlatformDisplayNames[toAppPlatform(profileInfo.platform)];
+
+      const applicationIdentifier = await getApplicationIdentifierAsync(
+        projectDir,
+        exp,
+        profileInfo.profile,
+        profileInfo.platform
+      );
+      const remoteVersions = await AppVersionQuery.latestVersionAsync(
+        projectId,
+        toAppPlatform(profileInfo.platform),
+        applicationIdentifier
+      );
+      const workflow = await resolveWorkflowAsync(projectDir, profileInfo.platform);
+      if (!remoteVersions?.buildVersion) {
+        Log.warn(
+          `Skipping versions sync for ${platformDisplayName}. There are no versions configured on Expo servers, use "eas build:version:set" or run a build to initialize it.`
+        );
+        continue;
+      }
+      if (workflow === Workflow.MANAGED) {
+        Log.warn(
+          `Skipping versions sync for ${platformDisplayName}. This command will have no effect on projects using managed workflow.`
+        );
```

Comment:
we should also print the current version information that we store on our service in case they want to do something with it. if they run this command they probably want this information.

maybe something like this? 

"The remote value for the iOS buildNumber is 42, but it was not synced to the local project. This command has no effect on projects using managed workflow."